instrumentation/reactor-netty-0.9/src/main/java/io/opentelemetry/javaagent/instrumentation/reactor_netty/ReactorHttpClientInstrumentationModule.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.reactor_netty;
+
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isStatic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import io.netty.bootstrap.Bootstrap;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.javaagent.instrumentation.api.CallDepthThreadLocalMap;
+import io.opentelemetry.javaagent.instrumentation.netty.v4_1.AttributeKeys;
+import io.opentelemetry.javaagent.tooling.InstrumentationModule;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import reactor.core.publisher.Mono;
+import reactor.netty.Connection;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.client.HttpClientRequest;
+
+@AutoService(InstrumentationModule.class)
+public final class ReactorHttpClientInstrumentationModule extends InstrumentationModule {
+
+  public ReactorHttpClientInstrumentationModule() {
+    super("reactor-httpclient");
```

Comment:
How about naming it `ReactorNettyInstrumentationModule`? And using the following instrumentation names: `"reactor-netty", "reactor-netty-0.9"` (naming in accordance with our discussion in https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1579)

Reply:
Yeah, it does seem logical, but... :) This is Reactor's HttpClient instrumentation, not reactor-netty-core. I think I would prefer to change module name to `reactor_httpclient` :)

Reply:
@mateuszrzeszutek @trask please help me with names :)

[Reactor Netty](https://projectreactor.io/docs/netty/release/reference/index.html) provides separate clients and servers for TCP, UDP and HTTP. Right now we instrument only HttpClient. I think we should leave module name as `reactor-netty`, because it may in the future collect instrumentation for all parts of it. But I don't know if this should be one module (and its name then should be `reactor-netty`) or it is better to have separate modules per client/server.

Reply:
> I think we should leave module name as `reactor-netty`

Definitely, I agree 👍 

> I don't know if this should be one module (and its name then should be reactor-netty) or it is better to have separate modules per client/server.

I think I would take the incremental approach: name the class `ReactorNettyInstrumentationModule` even if we only instrument HTTP right now; when we instrument TCP/UDP put them all in a single module; and if somebody wants to have them enabled/disabled separately then I'd split them into several `InstrumentationModule`s. 
Right now we're instrumenting a single jar/library so I think it makes sense to keep it in once module.